postgresql/postgresql_client.go

```diff
@@ -0,0 +1,167 @@
+
+	// Create the server
+	future, err := serversClient.Create(
+		ctx,
+		config.GroupName(),
+		serverName,
+		pg.Server{
+			Location: to.StringPtr(config.Location()),
+			Sku: &pg.Sku{
+				Name: to.StringPtr("Standard_D4s_v3"),
+				Tier: "GeneralPurpose",
+			},
+			ServerProperties: &pg.ServerProperties{
+				AdministratorLogin:         to.StringPtr(dbLogin),
+				AdministratorLoginPassword: to.StringPtr(dbPassword),
+				Version:                    pg.OneTwo,
+				StorageProfile: &pg.StorageProfile{
+					StorageMB: to.Int32Ptr(524288),
+				},
+			},
+		})
+
+	if err != nil {
+		return server, fmt.Errorf("cannot create pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return server, fmt.Errorf("cannot get the pg server create or update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// UpdateServerStorageCapacity given the server name and the new storage capacity it updates the server's storage capacity.
+func UpdateServerStorageCapacity(ctx context.Context, serversClient pg.ServersClient, serverName string, storageCapacity int32) (server pg.Server, err error) {
+
+	future, err := serversClient.Update(
+		ctx,
+		config.GroupName(),
+		serverName,
+		pg.ServerForUpdate{
+			ServerPropertiesForUpdate: &pg.ServerPropertiesForUpdate{
+				StorageProfile: &pg.StorageProfile{
+					StorageMB: &storageCapacity,
+				},
+			},
+		},
+	)
+	if err != nil {
+		return server, fmt.Errorf("cannot update pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return server, fmt.Errorf("cannot get the pg server update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// DeleteServer deletes the PostgreSQL server.
+func DeleteServer(ctx context.Context, serversClient pg.ServersClient, serverName string) (resp autorest.Response, err error) {
+
+	future, err := serversClient.Delete(ctx, config.GroupName(), serverName)
+	if err != nil {
+		return resp, fmt.Errorf("cannot delete the pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return resp, fmt.Errorf("cannot get the pg server update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// GetFwRulesClient returns the FirewallClient
+func GetFwRulesClient() pg.FirewallRulesClient {
+	fwrClient := pg.NewFirewallRulesClient(config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
+	fwrClient.Authorizer = a
+	fwrClient.AddToUserAgent(config.UserAgent())
+	return fwrClient
+}
+
+// CreateOrUpdateFirewallRule given the firewallname and new properties it updates the firewall rule.
+func CreateOrUpdateFirewallRule(ctx context.Context, fwrClient pg.FirewallRulesClient, serverName, firewallRuleName, startIPAddr, endIPAddr string) error {
+

```

Comment:
Extra empty line here, could we remove this?

Reply:
Fixed